core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -2151,16 +2151,16 @@ public boolean isTransactionRelevant(Transaction tx) throws ScriptException {
     }
 
     /**
-     * A transaction is mature if it is either a building coinbase tx that is as deep or deeper than the required coinbase depth, or a non-coinbase tx.
+     * Determine if a transaction is <i>mature</i>. A coinbase transaction is <i>mature</i> if it has been confirmed at least
+     * {@link NetworkParameters#getSpendableCoinbaseDepth()} times. On {@link BitcoinNetwork#MAINNET} this value is {@code 100}.
+     * For purposes of this method, non-coinbase transactions are also considered <i>mature</i>.
+     * @param tx The transaction to evaluate.
+     * @return {@code true} if it is a mature coinbase transaction or if it is not a coinbase transaction.
      */
     public boolean isTransactionMature(Transaction tx) {
-        if (!tx.isCoinBase())
-            return true;
-
-        if (tx.getConfidence().getConfidenceType() != ConfidenceType.BUILDING)
-            return false;
-
-        return tx.getConfidence().getDepthInBlocks() >= params.getSpendableCoinbaseDepth();
+        return !tx.isCoinBase() ||
+                tx.getConfidence().getConfidenceType() == ConfidenceType.BUILDING &&
+                tx.getConfidence().getDepthInBlocks() >= params.getSpendableCoinbaseDepth();
```

Comment:
I find this hard to read, because of

* the precedence rules of boolean algebra operators are often not present (at least not with me), and
* it reads less like a "rulebook", like the old variant.

But yeah. probably one must get used to it.

Reply:
In contrast, if we'd strip the
```
tx.getConfidence().getConfidenceType() == ConfidenceType.BUILDING &&
```
line because it's probably redundant, it would read fine.

Reply:
ok, I'll remove the `BUILDING` line. That will give you more time to get used to the "functional-style"...

Reply:
Fixed and force-pushed.